server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -702,14 +708,20 @@ public Node start() throws NodeValidationException {
 
         // Load (and maybe upgrade) the metadata stored on disk
         final GatewayMetaState gatewayMetaState = injector.getInstance(GatewayMetaState.class);
-        gatewayMetaState.start(
-            settings(),
-            transportService,
-            clusterService,
-            injector.getInstance(MetaStateService.class),
-            injector.getInstance(MetadataIndexUpgradeService.class),
-            injector.getInstance(MetadataUpgrader.class)
-        );
+        gatewayMetaState.start(settings(), transportService, clusterService, injector.getInstance(MetaStateService.class),
+                               injector.getInstance(MetadataIndexUpgradeService.class), injector.getInstance(MetadataUpgrader.class),
+                               injector.getInstance(PersistedClusterStateService.class));
```

Comment:
Maybe keep the one arg per line format. We've enough changes in this file already that it shouldn't have a big negative impact with further patches